pygeo/parameterization/DVGeoSketch.py

```diff
@@ -0,0 +1,296 @@
+    possible use most of the constraints DVConstraints since most of
+    those points lie on the surface.
+
+    2. It cannot handle *moving* intersection. A geometry with static
+    intersections is fine as long as the intersection doesn't move
+
+    3. It does not support complex numbers for the complex-step method.
+
+    4. It does not support separate configurations.
+
+    5. Because of limitations with ESP and OpenVSP, this class
+    uses parallel finite differencing to obtain the required Jacobian
+    matrices.
+
+    Parameters
+    ----------
+    fileName : str
+       filename of .vsp3 file (OpenVSP) or .csm file (ESP).
+
+    comm : MPI Intra Comm
+       Comm on which to build operate the object. This is used to
+       perform embarrassingly parallel finite differencing. Defaults to
+       MPI.COMM_WORLD.
+
+    scale : float
+       A global scale factor from the ESP/VSP geometry to incoming (CFD) mesh
+       geometry. For example, if the ESP/VSP model is in inches, and the CFD
+       in meters, scale=0.0254.
+
+    """
+
+    def __init__(self, fileName, comm=MPI.COMM_WORLD, scale=1.0, projTol=0.01):
+        super().__init__(fileName=fileName)
+
+        # this scales coordinates from model to mesh geometry
+        self.modelScale = scale
+        # and this scales coordinates from mesh to model geometry
+        self.meshScale = 1.0 / scale
+        self.projTol = projTol * self.meshScale  # default input is in meters.
+
+        self.updatedJac = {}
+        self.comm = comm
+
+        # Initial list of DVs
+        self.DVs = OrderedDict()
+
+    @abstractmethod
+    def addPointSet(self, points, ptName, **kwargs):
+        """
+        Add a set of coordinates to DVGeometry
+        The is the main way that geometry, in the form of a coordinate
+        list is given to DVGeometry to be manipulated.
+        """
+        pass
+
+    @abstractmethod
+    def setDesignVars(self, dvDict):
+        """
+        Standard routine for setting design variables from a design
+        variable dictionary.
+
+        Parameters
+        ----------
+        dvDict : dict
+            Dictionary of design variables. The keys of the dictionary
+            must correspond to the design variable names. Any
+            additional keys in the dv-dictionary are simply ignored.
+        """
+        pass
+
+    def getValues(self):
+        """
+        Generic routine to return the current set of design
+        variables. Values are returned in a dictionary format
+        that would be suitable for a subsequent call to setValues()
+
+        Returns
+        -------
+        dvDict : dict
+            Dictionary of design variables
+        """
+        dvDict = OrderedDict()
+        for dvName in self.DVs:
+            dvDict[dvName] = self.DVs[dvName].value
+
+        return dvDict
+
+    @abstractmethod
+    def update(self, ptSetName, config=None):
+        pass
+
+    def pointSetUpToDate(self, ptSetName):
```

Comment:
this function already exists in the parent class, why are you re-implementing it here?